src/vs/workbench/contrib/tasks/common/tasks.ts

```diff
@@ -1190,6 +1190,23 @@ export namespace KeyedTaskIdentifier {
 	}
 }
 
+export const enum TaskSettingId {
+	AutoDetect = 'task.autoDetect',
+	SaveBeforeRun = 'task.saveBeforeRun',
+	ShowDecorations = 'task.showDecorations'
+}
+
+export const enum TasksSettingId {
+	Tasks = 'tasks',
+	SuppressTaskName = 'tasks.suppressTaskName',
+	Windows = 'tasks.windows',
+	Osx = 'tasks.osx',
+	Linux = 'tasks.linux',
+	ShowOutput = 'tasks.showOutput',
+	IsShellCommand = 'tasks.isShellCommand',
+	ServiceTestSetting = 'tasks.service.testSetting',
+}
+
```

Comment:
I notice some settings aren't there:

<img width="325" alt="Screen Shot 2022-07-06 at 6 48 02 pm" src="https://user-images.githubusercontent.com/2193314/177672683-cd2a620d-a680-4166-a818-b069dff21012.png">

Mean to include these too? https://github.com/Microsoft/vscode/blob/7c78640d86e5de1ca0270912584d9d38beafeec1/src/vs/workbench/contrib/tasks/browser/task.contribution.ts#L428

Reply:
maybe should add those too - but i was just going for the ones that are referenced within the code base

Reply:
:+1: including everything means there's a single source of truth and strong typing where it's used